cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -162,6 +194,65 @@ public SystemPropertyModellingResult loadSystemPropertiesFromSource(SlangSource
         return extractProperties(parseModellingResult.getParsedSlang(), source, parseModellingResult.getErrors());
     }
 
+    private void handleStrategyAfterModelling(
+            SlangSource source,
+            PrecompileStrategy precompileStrategy,
+            String filePath,
+            ExecutableModellingResult executableModellingResult) {
+        switch (precompileStrategy) {
+            case WITH_CACHE:
+                cachedPrecompileService.cacheValue(filePath, executableModellingResult, source);
+                break;
+            case WITHOUT_CACHE:
+                break;
+            default:
+                throw new NotImplementedException(generatePreCompileTypeErrorMessage(precompileStrategy));
+        }
+    }
+
+    private CacheResult handleStrategyBeforeModelling(
+            SlangSource source,
+            PrecompileStrategy precompileStrategy,
+            String filePath) {
+        CacheResult cacheResult = null;
+        switch (precompileStrategy) {
+            case WITH_CACHE:
+                cacheResult = cachedPrecompileService.getValueFromCache(filePath, source);
+                break;
+            case WITHOUT_CACHE:
+                break;
+            default:
+                throw new NotImplementedException(generatePreCompileTypeErrorMessage(precompileStrategy));
```

Comment:
NPE would be thrown anyway from the `switch` - I added validation tot he method, also to the `Slang` interface 

👍